### PR TITLE
Adding github actions

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      run: composer validate
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:39]: maybe we should add `${{ matrix.php-versions }}` to the key too?

**Comment by JellyBellyDev** [comment]
I would also add the test of the build of the assets. WDYT?

**Comment by AlessandroMinoccheri** [comment]
Yes @JellyBellyDev , actually, this is the first tiny step.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - name: Install dependencies
+      uses: php-actions/composer@v5
+      with:
+        args: --prefer-dist
+        composer_version: 2
+        php_extensions: gd
+        php_version: 8.0
+
+    - name: Coding Standard Checks
+      run: PHP_CS_FIXER_IGNORE_ENV=1 ./bin/php-cs-fixer fix --dry-run -v
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:49]: `vendor/bin/php-cs-fixer fix --verbose --diff --dry-run` ;)

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        php-versions: [ '8.0' ]
+
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:16]:
```
    coverage-driver:          
        - "pcov"          
        - "xdebug3"
```

like here: https://github.com/sebastianbergmann/php-code-coverage/blob/master/.github/workflows/ci.yml#L55

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        php_version: 8.0
+
+    - name: Coding Standard Checks
+      run: PHP_CS_FIXER_IGNORE_ENV=1 ./bin/php-cs-fixer fix --dry-run -v
+
+    - name: PHPStan
+      run: ./vendor/bin/phpstan analyse --memory-limit=2G
+
+    - name: Test
+      run: ./bin/phpunit
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:55]:
```
- name: "Run tests with phpunit/phpunit"        
   run: "bin/phpunit -d memory_limit=-1 --coverage-clover clover.xml"
   
- name: "Send code coverage report to Codecov.io"        
   env:          
       CODECOV_TOKEN: "${{ secrets.CODECOV_TOKEN }}"        
   run: "bash &lt;(curl -s https://codecov.io/bash) || true"
```

File: tests/Controller/HomeControllerTest.php [hunk 5]
```diff
@@ -13,6 +13,7 @@ public function testHome(string $path): void
     {
         $client = self::createClient();
         $client->request('GET', $path);
+        var_dump($client->getResponse()->getContent());
```
Comment (JellyBellyDev) [inline tests/Controller/HomeControllerTest.php:16]: clean! ;)

File: .github/workflows/build.yml (hunk 4 above)
Comment (AlessandroMinoccheri) [reply .github/workflows/build.yml]: It doesn't work :(

**Comment by AlessandroMinoccheri** [comment]
I am not sure about uploading on codecov @JellyBellyDev

File: Makefile [hunk 6]
```diff
@@ -72,7 +72,7 @@ build_prod: ## build assets for production environment
 
 ##@ DEPLOY
 
-.docker_img_deps:
+	.docker_img_deps:
```
Comment (JellyBellyDev) [inline Makefile:75]: why you add a space?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        php-versions: [ '8.0' ]
+        coverage-driver: [ 'pcov', 'xdebug3' ]
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:16]: we want to leave only pcov?

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
+
+    - name: Install dependencies
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:49]: In the logs of this job i can see: `Warning: Unexpected input(s) 'composer_version', valid inputs are ['version', 'php_version', 'php_extensions', 'command', 'only_args', 'interaction', 'dev', 'progress', 'quiet', 'args', 'ssh_key', 'ssh_key_pub', 'ssh_domain', 'working_dir']`

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      with:
+        args: --prefer-dist
+        composer_version: 2
+        php_extensions: gd
+        php_version: 8.0
+
+    - name: Yarn install
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
```
Comment (JellyBellyDev) [inline .github/workflows/build.yml:60]: we can use `14.14.0` like in `.nvmrc` file?

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]
Please, can you override in the readme the badge for CI powered by github actions?

**Comment by AlessandroMinoccheri** [comment]
Readme updated @JellyBellyDev and did some fixes.

**Review by JellyBellyDev — APPROVED** [review]
LGTM 👍 

WDYT what do you think if we delete the contents of the dir `.circleci`?

It seems that only this is missing: https://github.com/PUGX/badge-poser/blob/master/.circleci/config.yml#L154

**Review by fabiocicerchia — APPROVED** [review]
👍 LGTM

**Comment by JellyBellyDev** [comment]
> LGTM 👍
> 
> WDYT what do you think if we delete the contents of the dir `.circleci`?
> 
> It seems that only this is missing: https://github.com/PUGX/badge-poser/blob/master/.circleci/config.yml#L154

i think it is better to remove .circle in a separate pr! ;)

**Pull request merged**